src/monattack.cpp

```diff
@@ -4720,6 +4726,142 @@ bool mattack::riotbot( monster *z )
     return true;
 }
 
+bool mattack::evolve_kill_strike( monster *z )
+{
+    Creature *target = z->attack_target();
+    if( target == nullptr ||
+        !is_adjacent( z, target, false ) ||
+        !z->sees( *target ) ) {
+        return false;
+    }
+    if( !z->can_act() ) {
+        return false;
+    }
+
+    z->moves -= 100;
+    bool uncanny = target->uncanny_dodge();
+    if( uncanny || dodge_check( z, target ) ) {
+        auto msg_type = target == &g->u ? m_warning : m_info;
+        target->add_msg_player_or_npc( msg_type, _( "The %s lunges at you, but you dodge!" ),
+                                       _( "The %s lunges at <npcname>, but they dodge!" ),
+                                       z->name() );
+        if( !uncanny ) {
```

Comment:
This will never trigger. This check should be outside of `if( uncanny || dodge_check( z, target ) )` check.

Reply:
Why not? It will be triggered when `uncanny` is false, but ` dodge_check( z, target )` yields true.

However: the message about lunging and dodging will be shown twice (once above and again inside this branch).